### PR TITLE
Add GetAllParams proto

### PR DESCRIPTION
**Pull request by helo9**
There is no possibility to get all parameters with MAVSDK at the moment. This PRs intend is prepare its implementation.

### CODE REVIEW

File: protos/param/param.proto [hunk 1]
```diff
@@ -33,6 +33,11 @@ service ParamService {
      * If the type is wrong, the result will be `WRONG_TYPE`.
      */
     rpc SetParamFloat(SetParamFloatRequest) returns(SetParamFloatResponse) { option (mavsdk.options.async_type) = SYNC; }
+    /*
+     * Get all parameters.
+     *
```
Comment (JonasVautherin) [inline protos/param/param.proto:38]:
```suggestion
```

File: protos/param/param.proto [hunk 2]
```diff
@@ -68,6 +73,27 @@ message SetParamFloatResponse {
     ParamResult param_result = 1;
 }
 
+message ParamListRequest {}
+
+message ParamListResponse {
+    AllParams params= 1;
```
Comment (JonasVautherin) [inline protos/param/param.proto:79]:
```suggestion
    AllParams params = 1;
```

**Review by JonasVautherin — CHANGES_REQUESTED** [review]
Yes I think that would work. You're missing the documentation, though (look at the comments elsewhere in the file, they are used to generate the documentation in the different languages :slightly_smiling_face:).

File: protos/param/param.proto [hunk 3]
```diff
@@ -68,6 +73,27 @@ message SetParamFloatResponse {
...
+
+message FloatParam {
+    string name = 1;
+    float value = 2;
+}
+
+message AllParams {
+    repeated IntParam int_params = 1;
+    repeated FloatParam float_params = 2;
+}
```
Comment (julianoes) [inline protos/param/param.proto:95]: @JonasVautherin should we consider protobuf maps for this?

https://developers.google.com/protocol-buffers/docs/proto#maps

Comment (JonasVautherin) [reply protos/param/param.proto:95]: Would be cool, but `protoc-gen-mavsdk` does not understand maps yet. So somebody would have to contribute that, and then the corresponding templates in all the languages. But that would be great, yes!

**Comment by helo9** [comment]
> Yes I think that would work. You're missing the documentation, though (look at the comments elsewhere in the file, they are used to generate the documentation in the different languages slightly_smiling_face).

I added documentation now.

**Review by JonasVautherin — COMMENTED** [review]
Great!

Before we merge this, I think the next step would be to generate the C++ files in mavlink/MAVSDK using this branch. Once we confirm it works in C++, we can merge this branch and the C++ implementation (and the language bindings will come for free).

Does that make sense?

**Comment by helo9** [comment]
I think the PR is ready, any concerns?

**Comment by JonasVautherin** [comment]
Yes, the C++ PR shows that this one lacks documentation (that's why the CI fails there :innocent:). Check my review here: https://github.com/mavlink/MAVSDK/pull/1191

**Comment by helo9** [comment]
> Yes, the C++ PR shows that this one lacks documentation (that's why the CI fails there innocent). Check my review here: [mavlink/MAVSDK#1191](https://github.com/mavlink/MAVSDK/pull/1191)

Apparently I missed that comment :blush:

Should be fixed now.

File: protos/param/param.proto [hunk 4]
```diff
@@ -68,6 +72,36 @@ message SetParamFloatResponse {
     ParamResult param_result = 1;
 }
 
+message GetAllParamsRequest {}
+
+message GetAllParamsResponse {
+    AllParams params = 1;
```
Comment (JonasVautherin) [inline protos/param/param.proto:78]:
```suggestion
    AllParams params = 1; // Collection of all parameters
```

**Review by JonasVautherin — APPROVED** [review]
LGTM, thanks!

**Pull request merged**